packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -90,6 +94,23 @@ export class CdkToolkit {
         throw new Error(`Stack ${stack.name} does not define an environment, and AWS credentials could not be obtained from standard locations or no region was configured.`);
       }
 
+      if (Object.keys(stack.template.Resources || {}).length === 0) { // The generated stack has no resources
+        const cfn = await options.sdk.cloudFormation(stack.environment.account, stack.environment.region, Mode.ForReading);
+        if (!await stackExists(cfn, stack.name)) {
+          warning('%s: stack has no resources, skipping deployment.', colors.bold(stack.name));
+        } else {
+          warning('%s: stack has no resources, skipping deployment and deleting existing stack.', colors.bold(stack.name));
+          await this.destroy({
+            stackNames: [stack.name],
+            exclusively: false,
```

Comment:
I think you want to pass `exclusively: true`, otherwise you're taking risks of possibly destroying a bunch of other stacks, too (depending what dependencies were recorded, possibly manually, on the resource-less stack).

Reply:
I'm confused about this. If other stacks depend on this stack then how can we destroy it? I don't understand this `exclusively` option...

Reply:
If the stack doesn't have any resources, other stacks can't depend on this stack nor it shouldn't be taking any dependencies (although technically the cloud assembly manifest can declare dependencies without any resources). I think @RomainMuller's point is for safety - `exclusively: false` means that any stacks that this stack depends on will also be destroyed which is *not* what we want. So making this `true` is the safer choice.